static/app/components/events/interfaces/frame/utils.tsx

```diff
@@ -149,3 +150,92 @@ export function hasFileExtension(filepath: string) {
   // Check if the filepath matches the pattern
   return fileExtensionPattern.test(filepath);
 }
+
+/**
+ * Extracts the origin URL from an event
+ *
+ * TODO: Should consider other sources of origin besides just url tag
+ *
+ * @param event The event to extract the origin from
+ * @returns The origin URL string, or empty string if not found/invalid
+ */
+function extractEventOrigin(event: Event): string {
+  const urlTag = event.tags.find(({key}) => key === 'url');
+
+  if (!urlTag?.value) {
+    return '';
+  }
+
+  try {
+    const url = new URL(urlTag.value);
+    return url.origin;
+  } catch {
+    return '';
+  }
```

Comment:
we do have [this utility function](https://github.com/getsentry/sentry/blob/master/static/app/utils/url/safeURL.tsx#L5) that helps with some of this try catch repitition
```suggestion
  return safeURL(urlTag.value)?.origin ?? '';
```